FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_plain_text/DemoTasks/PlaintextMQTTExample.c

```diff
@@ -343,6 +385,18 @@ static void prvMQTTDemoTask( void * pvParameters )
         xMQTTStatus = MQTT_ProcessLoop( &xMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
         configASSERT( xMQTTStatus == MQTTSuccess );
 
+        /* Check if recent subscription request has been rejected. globalSubAckStatus is updated
+         * in eventCallback to reflect the status of the SUBACK sent by the broker. */
+        if( pxGlobalSubAckStatus[ 0 ] == MQTTSubAckFailure )
+        {
+            /* If server rejected the subscription request, attempt to resubscribe to topic.
+             * Attempts are made according to the exponential backoff retry strategy
+             * implemented in retryUtils. */
+            LogInfo( ( "Server rejected initial subscription request. Attempting to re-subscribe to topic %s.",
+                       mqttexampleTOPIC ) );
+            prvHandleResubscribe( &xMQTTContext );
```

Comment:
So, this is not the same function used to subscribe the first time...  does that mean one has backoff and the other does not?  Would it be better to have a single function that works in both cases?

Reply:
Good suggestion, I have made this to a single function. Thanks